app/frontend/Shared/ContextMenu.ts

```diff
@@ -283,3 +287,32 @@ export function download(url: URLString, window: any, howSaveAsDialog: boolean,
   a.click();
   a.href = "";
 }
+// #else
+// @ts-ignore ts2300
+export async function download(url: URLString, win: any, howSaveAsDialog: boolean, filename?: string) {
+  let urlObj = new URL(url);
+  let blob: Blob;
+  if (urlObj.protocol == "http:" || urlObj.protocol == "https:") {
+    let res = await fetch(urlObj.href);
+    filename = urlObj.pathname.split("/").pop();
+    blob = await res.blob();
+  } else if (urlObj.protocol == "data:") {
+    blob = await dataURLToBlob(url);
+  } else {
+    throw new Error("Unsupported protocol: " + urlObj.protocol);
+  }
+  let ext = fileExtensionForMIMEType(blob.type);
+  filename = sanitize.filename(filename, `${filename ?? "untitled"}.${ext}`);
+  if (howSaveAsDialog) {
+    let a = document.createElement("a");
+    a.href = url;
+    a.setAttribute("target", "_blank"); // for non-same-origin images
+    a.setAttribute("download", filename);
+    a.click();
+    a.href = "";
+  } else {
```

Comment:
Do we need this? Why do we need both "Save image" and "Save image as..."? Isn't one sufficient? I'd keep "Save image as..." (1), but rename it to "Save image…" (with the … ellipsis)

(1) given that it's simpler to implement, and more helpful to select the file location

Reply:
If you keep only "save image as..." (= `howSaveAsDialog = true`), then do we still need 2 different download() function implementations? Do you still need to convert to blob?

Actually, you are making a lot of effort to convert to blob, and then you don't use it at all in the `howSaveAsDialog = true` case.